### PR TITLE
Vulkan: Initial swapchain implementation

### PR DESCRIPTION
**Pull request by Kangz**
PTAL!

### CODE REVIEW

File: src/backend/vulkan/TextureVk.cpp [hunk 1]
```diff
@@ -96,8 +96,14 @@ namespace backend { namespace vulkan {
                         VK_ACCESS_COLOR_ATTACHMENT_READ_BIT | VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT;
                 }
             }
-
-            // TODO(cwallez@chromium.org): What about present? Does it require VK_MEMORY_READ_BIT?
+            if (usage & nxt::TextureUsageBit::Present) {
+                // There is no access flag for present because the VK_KHR_SWAPCHAIN extension says
+                // that vkQueuePresentKHR makes the memory of the image visible to the presentation
+                // engine. There's also a not explicitly saying this is what dstAccessMask should be
```
Comment (kainino0x) [inline src/backend/vulkan/TextureVk.cpp]: "not explicitly" -> "note explicitly"?
"this is what dstAccessMask should be 0" -> ?

File: src/backend/vulkan/NativeSwapChainImplVk.cpp [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+                                           VK_PIPELINE_STAGE_TOP_OF_PIPE_BIT, 0, 0, nullptr, 0,
+                                           nullptr, 1, &barrier);
+        }
+
+        return NXT_SWAP_CHAIN_NO_ERROR;
+    }
+
+    nxtSwapChainError NativeSwapChainImpl::GetNextTexture(nxtSwapChainNextTexture* nextTexture) {
+        // Transiently create a semaphore that will be signaled when the presentation engine is done
+        // with the swapchain image. Further operations one the image will wait for this semaphore.
```
Comment (kainino0x) [inline src/backend/vulkan/NativeSwapChainImplVk.cpp]: one the image -> on the image

Comment (Kangz) [reply src/backend/vulkan/NativeSwapChainImplVk.cpp]: Done

File: src/backend/vulkan/TextureVk.cpp (hunk 1 above)
Comment (Kangz) [reply src/backend/vulkan/TextureVk.cpp]: Done

**Pull request merged**